### PR TITLE
Add function bitset orientation compatible patches

### PR DESCRIPTION
**Pull request by sloriot**
## Summary of Changes

More complete version of `merge_reversible_connected_components()` that returns a vector with compatible orientation of the different connected components. The reversal is left to the user depending on its constraints.

## Release Management
* Affected package(s): PMP
* Issue(s) solved (if any): 
* Feature/Small Feature (if any): [here](https://cgal.geometryfactory.com/CGAL/Members/wiki/Features/Small_Features/PMP_Orient_bits)
* Link to compiled documentation (obligatory for small feature) [here](https://cgal.github.io/6212/v0/Polygon_mesh_processing/group__PMP__orientation__grp.html#ga6538c18815e4e412e1f6002ebc76f833)
* License and copyright ownership: GPL3/GF

### CODE REVIEW

**Comment by sloriot** [comment]
The documentation is available, after a few minutes, here: https://cgal.github.io/6212/v0/Polygon_mesh_processing/index.html

File: Polygon_mesh_processing/examples/Polygon_mesh_processing/cc_compatible_orientations.cpp [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+{
+// create a mesh with many connected connected components that should
+// be reoriented to define a valid closed mesh
+  const std::string filename = CGAL::data_file_path("meshes/elephant.off");
+
+  Mesh mesh;
+  CGAL::IO::read_polygon_mesh(filename, mesh);
+
+  // turn the mesh into a triangle soup, duplicating all the vertices and shuffling orientations
+  std::vector<Point> points;
```
Comment (lrineau) [inline Polygon_mesh_processing/examples/Polygon_mesh_processing/cc_compatible_orientations.cpp:33]: The indentation of comments is not always align with the indentation of the code. Is that on purpose?

File: Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/orientation.h [hunk 2]
```diff
@@ -1604,6 +1604,236 @@ void merge_reversible_connected_components(PolygonMesh& pm,
...
+ *     \cgalParamDefault{`boost::get(CGAL::vertex_point, pm)`}
+ *     \cgalParamExtra{If this parameter is omitted, an internal property map for `CGAL::vertex_point_t`
+ *                     should be available for the vertices of `pm`.}
+ *   \cgalParamNEnd
+ * \cgalNamedParamsEnd
+ */
+template <class PolygonMesh, class FaceBitMap, class NamedParameters = parameters::Default_named_parameters>
+bool connected_components_compatible_orientations(PolygonMesh& pm,
+                                                  FaceBitMap fbm,
+                                                  const NamedParameters& np = parameters::default_values())
```
Comment (lrineau) [inline Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/orientation.h:1642]: The documentation should explain why `pm` is take by reference to non-const. When I read the documentation, I understand that the mesh is not modified (only property maps are), but when I read the signature of the function template, then I have a doubt.

File: Polygon_mesh_processing/examples/Polygon_mesh_processing/cc_compatible_orientations.cpp (hunk 1 above)
Comment (sloriot) [reply Polygon_mesh_processing/examples/Polygon_mesh_processing/cc_compatible_orientations.cpp:33]: Kind of, the one without leading space are the big steps, the other are intermediate comments. I don't really know how to do better, putting spaces after `//`

Comment (lrineau) [reply Polygon_mesh_processing/examples/Polygon_mesh_processing/cc_compatible_orientations.cpp:33]: If that is one purpose, then that is fine for me. I close this discussion.

File: Polygon_mesh_processing/examples/Polygon_mesh_processing/cc_compatible_orientations.cpp [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+  }
+
+  // load the soup into the mesh;
+  mesh.clear();
+  PMP::polygon_soup_to_polygon_mesh(points, triangles, mesh);
+  std::ofstream("soup.off") << mesh;
+
+// now check how face orientation should be reversed
+  auto fbm = mesh.add_property_map<Mesh::Face_index, bool>("fbm", false).first;
+  PMP::connected_components_compatible_orientations(mesh, fbm);
```
Comment (MaelRL) [inline Polygon_mesh_processing/examples/Polygon_mesh_processing/cc_compatible_orientations.cpp:57]: Check the return?

File: Polygon_mesh_processing/examples/Polygon_mesh_processing/cc_compatible_orientations.cpp [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+  for (Mesh::Face_index f : mesh.faces())
+  {
+    Mesh::Halfedge_index h = mesh.halfedge(f);
+    std::size_t s = points.size();
+    points.push_back(mesh.point(source(h,mesh)));
+    points.push_back(mesh.point(target(h,mesh)));
+    points.push_back(mesh.point(target(mesh.next(h),mesh)));
+    triangles.push_back( {s, s+1, s+2} );
+    std::shuffle(triangles.back().begin(), triangles.back().end(), g);
+  }
```
[the remark this answers is not in the document]
Comment (sloriot) [reply Polygon_mesh_processing/examples/Polygon_mesh_processing/cc_compatible_orientations.cpp:48]: Note that vertices are all duplicated so reading as a soup won't help much.

File: Polygon_mesh_processing/examples/Polygon_mesh_processing/cc_compatible_orientations.cpp (hunk 3 above)
Comment (sloriot) [inline Polygon_mesh_processing/examples/Polygon_mesh_processing/cc_compatible_orientations.cpp:57]:
```suggestion
  bool is_orientable = PMP::connected_components_compatible_orientations(mesh, fbm);
  assert(is_orientable);
```

File: Polygon_mesh_processing/examples/Polygon_mesh_processing/cc_compatible_orientations.cpp [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+  PMP::polygon_soup_to_polygon_mesh(points, triangles, mesh);
+  std::ofstream("soup.off") << mesh;
+
+// now check how face orientation should be reversed
+  auto fbm = mesh.add_property_map<Mesh::Face_index, bool>("fbm", false).first;
+  PMP::connected_components_compatible_orientations(mesh, fbm);
+
+// now reverse orientation of faces with bit 1
+  std::vector<Mesh::Face_index> faces_to_reverse;
+  for (Mesh::Face_index f : mesh.faces())
```
[the remark this answers is not in the document]
Comment (sloriot) [reply Polygon_mesh_processing/examples/Polygon_mesh_processing/cc_compatible_orientations.cpp:63]: I don't want to change it.

**Comment by lrineau** [comment]
I have resolved the merge conflict, so that the CI actions could be tested again: https://github.com/CGAL/cgal/pull/6212/commits/a4f4c276f312ad4e8dcb4f7d827cf07b8f51e1a1.

**Comment by sloriot** [comment]
\build:v1

**Comment by sloriot** [comment]
Successfully tested in CGAL-5.5-Ic-60

**Pull request merged**